### PR TITLE
set defaults for -DCASA_BUILD=1 (i.e. building within CASA context)

### PR DESCRIPTION
**Pull request by schiebel**
Changes to set defaults for CASA developers when building casacore within a larger CASA build.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -83,6 +83,37 @@ if( CASA_BUILD )
...
+      ### RHEL7
+      set(PYTHON_LIBRARY "/opt/casa/01/lib/libpython2.7.so")
+      set(PYTHON_INCLUDE_DIR "/opt/casa/01/include/python2.7")
+      set(PYTHON_EXECUTABLE:FILEPATH "/opt/casa/01/bin/python")
+      if (EXISTS "/usr/include/boost")
+         set(BOOST_ROOT "/usr")
+      endif( )
+   elseif(EXISTS "/usr/lib64/casa/01/include/python2.7")
+      ### RHEL5/RHEL6
+      set(ENV{PATH} "/usr/lib64/casa/01/bin" $ENV{PATH})
```
Comment (juliantaylor) [inline CMakeLists.txt]: this doesn't make much sense
environment variables have no influence after cmake ends and the setting of absolute paths should already setup the requited binaries

though SWIG_EXECUTABLE is missing

as written this probably breaks the configure as it overrides the PATH completely. or does cmake to something special here?

**Comment by tammojan** [comment]
About `Boost_NO_BOOST_CMAKE`: you need to set this because casa accidentally bundles CMake config files, which it shouldn't. In particular, the following files (from casa release 4.6) should not be bundled:
```
lib/boost/Boost.cmake
lib/boost/BoostConfig.cmake
lib/boost/BoostConfigVersion.cmake
lib/boost/Boost-relwithdebinfo.cmake
```
These files contain hardcoded paths to where boost happened to be on the casa build machine.

File: CMakeLists.txt (hunk 1 above)
Comment (schiebel) [inline CMakeLists.txt]: I was trying to have cmake find the right exectuables as part of it's configuration phase but did not want users to have to configure their path outside of cmake... I am not completely sure that this did achieve that goal, but this was the reference I used:

http://stackoverflow.com/questions/21047399/cmake-set-environment-variables-from-a-script

Comment (juliantaylor) [inline CMakeLists.txt]: configuring the PATH should not be necessary for casacore, it only needs python which is already defined

also forget about my swig comment, I forgot thats only needed for gcwrap not in casacore

**Comment by schiebel** [comment]
Thanks @tammojan , I assume you mean 4.5 release. It is true that this was a bit clunky as it included whole RPM distros. This has been resolved with the 4.6 prerelease (https://svn.cv.nrao.edu/casa/distro/linux/prerelease/el6/casa-prerelease-4.6.0-14.tar.gz). However, this is a separate issue. Boost_NO_BOOST_CMAKE was needed because of errors in the boost cmake configuration files on RHEL6 (see below).

Also @tammojan , please check https://travis-ci.org/casacore/casacore/jobs/119542326 ; it seems like either the tests sometimes run to long or there is a configuration issue for osx.

thanks very much!

RHEL6 error without Boost_NO_BOOST_CMAKE
-------------------------------------------------------------------------------
`
-bash-4.1$ rm -rf * && cmake -DCASA_BUILD=ON -DBUILD_TESTING=OFF '-DCMAKE_INSTALL_PREFIX=/tmp/linux' -DCMAKE_BUILD_TYPE=Debug ..
-- The C compiler identification is GNU 4.9.2
-- The CXX compiler identification is GNU 4.9.2
-- Check for working C compiler: /opt/rh/devtoolset-3/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-3/root/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /opt/rh/devtoolset-3/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-3/root/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- The Fortran compiler identification is GNU
-- Check for working Fortran compiler: /opt/rh/devtoolset-3/root/usr/bin/gfortran
-- Check for working Fortran compiler: /opt/rh/devtoolset-3/root/usr/bin/gfortran  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /opt/rh/devtoolset-3/root/usr/bin/gfortran supports Fortran 90
-- Checking whether /opt/rh/devtoolset-3/root/usr/bin/gfortran supports Fortran 90 -- yes
-- Shared object version number 1125.57.69
-- Performing Test HAS_CXX11
-- Performing Test HAS_CXX11 - Success
-- Performing Test HAS_GXX_FORTRAN_RULES
-- Performing Test HAS_GXX_FORTRAN_RULES - Success
-- Performing Test HAS_GCC_FORTRAN_RULES
-- Performing Test HAS_GCC_FORTRAN_RULES - Success
-- Looking for pread
-- Looking for pread - found
-- Looking for pwrite
-- Looking for pwrite - found
-- Found Readline: /usr/lib64/libreadline.so  
-- Could NOT find SOFA (missing:  SOFA_LIBRARY) 
-- Found FLEX: /usr/bin/flex  
-- Found BISON: /usr/bin/bison  
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - found
-- Looking for include file pthread.h
-- Looking for include file pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- A library with BLAS API found.
-- A library with BLAS API found.
-- Looking for Fortran cheev
-- Looking for Fortran cheev - found
-- A library with LAPACK API found.
-- Found FFTW3: /usr/lib64/libfftw3f_threads.so  
-- Found CFITSIO: /usr/lib64/libcfitsio.so (found suitable version "3.350", minimum required is "3.030") 
-- Found WCSLIB: /usr/lib64/libwcs.so (found suitable version "4.7", minimum required is "4.7") 
-- Found Pthreads: /usr/lib64/libpthread.so  
CMake Warning at measures/CMakeLists.txt:8 (message):
  No ephemerides data found in the specified DATA_DIR ("%CASAROOT%/data"), so
  the location must be specified at runtime.  If you have preinstalled
  casacore data, specify the location with -DDATA_DIR.


-- Looking for python2 specific environment...
-- Found PythonInterp: /usr/lib64/casa/01/bin/python2 (found suitable version "2.7.5", minimum required is "2") 
-- Found PythonLibs: /usr/lib64/casa/01/lib/libpython2.7.so (found suitable version "2.7.5", minimum required is "2") 
CMake Error at /usr/lib64/boost/Boost.cmake:536 (message):
  The imported target "boost_date_time-static" references the file

     "/usr/lib64/lib64/libboost_date_time.a"

  but this file does not exist.  Possible reasons include:

  * The file was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and contained

     "/usr/lib64/boost/Boost.cmake"

  but not all the files it references.

Call Stack (most recent call first):
  /usr/lib64/boost/BoostConfig.cmake:28 (include)
  /usr/share/cmake/Modules/FindBoost.cmake:177 (find_package)
  python/CMakeLists.txt:25 (find_package)


-- Configuring incomplete, errors occurred!
See also ".../casacore.mine/build/CMakeFiles/CMakeOutput.log".
See also ".../casacore.mine/build/CMakeFiles/CMakeError.log".
-bash-4.1$ uname -a
Linux xxxxx 2.6.32-573.22.1.el6.x86_64 #1 SMP Thu Mar 17 03:23:39 EDT 2016 x86_64 x86_64 x86_64 GNU/Linux
-bash-4.1$ 

`

**Comment by schiebel** [comment]
@juliantaylor seems reasonable, I'll submit another pull request shortly...

**Comment by tammojan** [comment]
> Also @tammojan , please check https://travis-ci.org/casacore/casacore/jobs/119542326 ; it seems like either the tests sometimes run too long or there is a configuration issue for osx.

@schiebel Sometimes the OSX jobs time out, there's not much we can do to change that. I'd rather leave the builds, and accept an occasional false positive from travis.

@schiebel Yes I meant 4.5. Good to hear that the Boost.cmake was removed from 4.6!

**Pull request closed without merging**